ssl/statem/statem_srvr.c

```diff
@@ -1960,9 +1960,8 @@ static int tls_early_post_process_client_hello(SSL_CONNECTION *s)
     }
 
     if (!s->hit
-            && s->version >= TLS1_VERSION
-            && !SSL_CONNECTION_IS_TLS13(s)
-            && !SSL_CONNECTION_IS_DTLS(s)
+            && ssl_version_cmp(s, s->version, SSL_CONNECTION_IS_DTLS(s) ? DTLS1_VERSION : TLS1_VERSION) >= 0
```

Comment:
Previously this did not allow DTLS at all, now it allows DTLS 1 or 1.2 - is this intentional?

Reply:
No I don't think so. Good catch!